src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -335,6 +341,11 @@ protected override void Dispose(bool disposing)
 
         private void OnDependenciesChanged(object sender, DependenciesChangedEventArgs e)
         {
+            lock(_lockDataSourceVersions)
+            {
+                _requestedTreeUpdateDataSources.Add(e.DataSourceVersions);
```

Comment:
why don't i just keep one merged dictionary with all datasources with max versions i received and always send that one?